### PR TITLE
Add endpoint for transaction costs report

### PR DESCRIPTION
**Pull request by mishra-ankit**
- Added protected endpoint `donations/transaction_costs_report`
Implements - https://github.com/stiftelsen-effekt/effekt-admin/issues/208

### CODE REVIEW

File: src/custom_modules/DAO_modules/donations.ts [hunk 1]
```diff
@@ -158,6 +158,37 @@ async function getAll(sort, page, limit = 10, filter = null) {
   }
 }
 
+async function getTransactionCostsReport() {
+    let [res] = await DAO.query(`
+    SELECT count(ID) as donationsCount,
+      (SELECT round(sum(transaction_cost)) FROM Donations
+        WHERE YEAR(timestamp_confirmed) = YEAR(CURRENT_DATE)
```
Comment (xim) [inline src/custom_modules/DAO_modules/donations.ts:165]: `BETWEEN LAST_DAY(CURRENT_TIMESTAMP - INTERVAL 2 MONTH) + INTERVAL 1 DAY AND LAST_DAY(CURRENT_TIMESTAMP - INTERVAL 1 MONTH)`

**Review by fellmirr — APPROVED** [review]
Looks solid :+1:

File: src/custom_modules/DAO_modules/donations.ts [hunk 2]
```diff
@@ -158,6 +158,37 @@ async function getAll(sort, page, limit = 10, filter = null) {
...
+        AND MONTH(timestamp_confirmed) = MONTH(CURRENT_DATE) - 1) as costPrevMonth,
+      (SELECT round(sum(transaction_cost)) FROM Donations
+        WHERE YEAR(timestamp_confirmed) = YEAR(CURRENT_DATE) - 1
+        AND MONTH(timestamp_confirmed) = MONTH(CURRENT_DATE) - 1) as costPrevMonthPrevYear,
```
Comment (xim) [inline src/custom_modules/DAO_modules/donations.ts:169]: `BETWEEN LAST_DAY(CURRENT_TIMESTAMP - INTERVAL 1 YEAR - INTERVAL 2 MONTH) + INTERVAL 1 DAY AND LAST_DAY(CURRENT_TIMESTAMP - INTERVAL 1 YEAR - INTERVAL 1 MONTH)`

**Review by xim — COMMENTED** [review]
I think my changes correctly handle January

**Comment by mishra-ankit** [comment]
> I think my changes correctly handle January

To keep UI design simple we decided to have not data in "Previous Month" during January. If that is what we're attempting to fix with the query ?
https://github.com/stiftelsen-effekt/effekt-admin/issues/208#issuecomment-1327297971

**Comment by xim** [comment]
> > I think my changes correctly handle January
> 
> To keep UI design simple we decided to have no data (NA) in "Previous Month" during January. If that is what we're attempting to fix with the query ? [stiftelsen-effekt/effekt-admin#208 (comment)](https://github.com/stiftelsen-effekt/effekt-admin/issues/208#issuecomment-1327297971)

`MONTH(timestamp_confirmed) = MONTH(CURRENT_DATE) - 1)` will query for `12 = 0` when the current month is January and the previous one is December

**Comment by mishra-ankit** [comment]
> > > I think my changes correctly handle January
> > 
> > 
> > To keep UI design simple we decided to have no data (NA) in "Previous Month" during January. If that is what we're attempting to fix with the query ? [stiftelsen-effekt/effekt-admin#208 (comment)](https://github.com/stiftelsen-effekt/effekt-admin/issues/208#issuecomment-1327297971)
> 
> `MONTH(timestamp_confirmed) = MONTH(CURRENT_DATE) - 1)` will query for `12 = 0` when the current month is January and the previous one is December

That should return null during January, as expected.

**Pull request merged**